### PR TITLE
Trimming and NativeAOT compatible

### PR DESCRIPTION
**Pull request by hez2010**
Making TextMateSharp trimming and NativeAOT compatible.

image

### CODE REVIEW

File: src/TextMateSharp.Grammars/RegistryOptions.cs [hunk 1]
```diff
@@ -171,6 +183,13 @@ void InitializeAvailableGrammars()
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
Comment (hez2010) [inline src/TextMateSharp.Grammars/RegistryOptions.cs:186]: This method will be inlined by the compiler so it won't affect the performance at all.

**Comment by danipen** [comment]
@hez2010 first of all thanks for the contribution. 

Question: do we need to change something in the NuGet package generation?

**Comment by danipen** [comment]
I'm just wondering just for clarity when reading the code, if we can set the attribute `DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)` in each class itself (`GrammarDefinition`, `Repository`, `Contributes`, etc ... ) instead of defining the `KeepType` in the `InitializeAvailableGrammars`.

What do you think?

**Comment by hez2010** [comment]
> I'm just wondering just for clarity when reading the code, if we can set the attribute `DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)` in each class itself (`GrammarDefinition`, `Repository`, `Contributes`, etc ... ) instead of defining the `KeepType` in the `InitializeAvailableGrammars`.
> 
> What do you think?

> Unfortunately, `DynamicallyAccessedMembers` doesn't work in that way.

**Comment by danipen** [comment]
Sorry I'm getting this in the demo application when testing you PR :-\ Not sure what's wrong.
image

**Comment by hez2010** [comment]
> Sorry I'm getting this in the demo application when testing you PR :-\ Not sure what's wrong. image

It seems that you are running as a x86 process? I changed the default architecture to x64 on Windows.

**Comment by danipen** [comment]
Also, it seems that the tests are hanging due to the x68/x64 issue.

**Comment by danipen** [comment]
Now all passed green!

**Comment by danipen** [comment]
Well I'm running "Debug" and "Any CPU" environments, but it seems that VS wants x86 not sure why, let's keep this for compatibility since x86 is compatible with both but the other way around ;-)

**Comment by danipen** [comment]
> Unfortunately, DynamicallyAccessedMembers doesn't work in that way.

No worries, we can leave as it.

**Comment by hez2010** [comment]
To test NativeAOT, you can change the demo project to net7.0 and use `dotnet publish -c Release -r win-x64 /p:PublishAot=true`. Note that by default the x86 onigwrap.dll will be copied to the publish directory and results in architecture mismatch, you need to copy the x64 onigwrap.dll to the publish directory manually. (This only happens in the development environment and won't affect the published nuget package).

**Comment by danipen** [comment]
> But I'm thinking that maybe we can reimplement the searcher using built-in Regex instead of oniguruma in the future?

It would be great to have 100% managed code and get rid of native deps!

I'm not sure how hard it will be. Currently, [grammar definition files](https://github.com/danipen/TextMateSharp/tree/master/src/TextMateSharp.Grammars/Resources/Grammars) (which have been taken from [Visual Studio code](https://github.com/microsoft/vscode/tree/main/extensions)) talk Oniguruma dialect, so I don't know how hard would be to write a mechanism that converts those regexes to something understandable by C# built-in regex syntax.

**Pull request merged**